src/data-mgt/node/utils/transform.js

```diff
@@ -215,39 +215,32 @@ const transform = {
     }
   },
 
-  transformMeasurement: async (measurement) => {
+  transformMeasurement: (measurement) => {
     try {
       const deviceCategory = measurement.field9
         ? measurement.field9
         : "lowcost";
-
-      let newObj = await Object.entries(measurement).reduce(
-        (newObj, [field, value]) => {
-          if (value) {
-            let transformedField = "";
-            if (deviceCategory === "reference") {
-              logText("the device is a BAM");
-              transformedField = transform.getBamFieldLabel(field);
-              logElement("transformedField", transformedField);
-            } else if (deviceCategory === "lowcost") {
-              logText("the device is a lowcost one");
-              transformedField = transform.getFieldLabel(field);
-            } else {
-              logText("the device does not have a category/type");
-              return {};
-            }
-            return {
-              ...newObj,
-              [transformedField]: value,
-            };
-          }
-        },
-        {}
-      );
-      delete newObj["undefined"];
-      return cleanDeep(newObj);
+      let response = {};
```

Comment:
just utilised a new implementation for measurement transformation. Previous one with JS array method `reduce` was somehow buggy